exporter/signalfxexporter/translation/delta_translator.go

```diff
@@ -72,14 +72,22 @@ func (t *deltaTranslator) deltaPt(deltaMetricName string, currPt *sfxpb.DataPoin
 }
 
 func doubleDeltaPt(currPt *sfxpb.DataPoint, prevPt *sfxpb.DataPoint, deltaMetricName string) *sfxpb.DataPoint {
+	delta := *currPt.Value.DoubleValue - *prevPt.Value.DoubleValue
+	if delta < 0 {
+		return nil
+	}
 	deltaPt := basePt(currPt, deltaMetricName)
-	*deltaPt.Value.DoubleValue = *currPt.Value.DoubleValue - *prevPt.Value.DoubleValue
+	*deltaPt.Value.DoubleValue = delta
 	return deltaPt
 }
 
 func intDeltaPt(currPt *sfxpb.DataPoint, prevPt *sfxpb.DataPoint, deltaMetricName string) *sfxpb.DataPoint {
+	delta := *currPt.Value.IntValue - *prevPt.Value.IntValue
+	if delta < 0 {
+		return nil
```

Comment:
on second though, it might be better to return `currPt.Value.IntValue` instead of `nil`. what do you think? it should be closer to the correct signal than skipping it. I'm wondering what is done in signalfx-agent in this scenario? sorry for causing the confusion.

Reply:
Skipping the datapoint altogether feels closer to being accurate since we don't have data, but I don't have a strong opinion on the matter. @bogdandrutu or @keitwb, do you have any thoughts on this?

Reply:
Yeah it seems best to just return a datapoint of 0 so that the timeseries is continuous without big gaps.

Reply:
The smart agent doesn't really deal with this as we use cumulative and gauges for almost everything.

Definitely better to keep a continuous stream of datapoints even if there are 0s.

Reply:
I agree with the continuous stream, but don't understand why returning `0` would give us a result closer to being accurate. If we do that, it means that we delay the known delta (presumably reported after a reset) from pushing now, and move it to the next datapoint instead. So we would have the following reports:
now: 0
next datapoint: `currPt.Value.IntValue` + `next delta`

I'd still suggest that we return `currPt.Value.IntValue` instead of `0`, then we would have smother stream with:
now: `currPt.Value.IntValue`
next datapoint: `next delta`

Let me know if it makes sense

Reply:
Sorry, I meant 0 or < 0 (negative number).  Is the "reset" scenario the only possibility for a counter being lower like this?  E.g. is it possible that a metric emitter might be trying to "correct" things by sending a lower number?   If "reset" is the only possibility then what you are saying makes sense to use the current value and then next delta.  If it is not due to a "reset" though it would be a corruption of the time series data to use so this needs to be considered carefully.

Reply:
Thanks for the explanations. Looks like if the timeseries is corrupted we'll get a blip, but not sure if there's a better way to handle it. I've update the PR to emit the last value if the delta is negative.

Reply:
We assume that it's a cumulative metric. If the source of the metrics is external like prometheus scraping, we can expect resets. I don't think we should handle corrupted data, for those cases it's probably acceptable to have the blips.